src/libs/string.liq

```diff
@@ -381,3 +381,21 @@ let string.char.ascii.number = list.init(10, fun (c) -> c + 48)
 def string.char.ascii.random(range=[...string.char.ascii]) =
   string.char(list.nth(range, random.int(min=0, max=list.length(range) - 1)))
 end
+
+# Escape HTML entities.
+# @category String
+def string.escape.html(s) =
+  string.replace(
+    pattern = "[&<>\"']",
+    fun (c) -> begin
+      if c == "&" then "&amp;"
+      elsif c == "<" then "&lt;"
+      elsif c == ">" then "&gt;"
+      elsif c == "\"" then "&quot;"
+      elsif c == "'" then "&apos;"
+      else c
+      end
+    end,
+    s
+  )
```

Comment:
```suggestion
  r/[&<>"']/g.replace(
    fun (c) -> begin
      if c == "&" then "&amp;"
      elsif c == "<" then "&lt;"
      elsif c == ">" then "&gt;"
      elsif c == '"' then "&quot;"
      elsif c == "'" then "&#39;"
      else c
      end
    end,
    s
  )
```
I'd prefer to use [`r/_/g.replace(fn, str)`](https://www.liquidsoap.info/doc-dev/language.html#regular-expressions) regular expression syntax.

Reply:
the `&#` aproach is actually shorter to implement if I can get the char code in any way (and would use it for all of them in that case), but could not find a way to do that. Is there any?